source/image/ImageSet.cpp

```diff
@@ -24,76 +25,73 @@ this program. If not, see <https://www.gnu.org/licenses/>.
 
 #include <algorithm>
 #include <cassert>
-#include <iterator>
 
 using namespace std;
 
 namespace {
-	// Determine whether the given path is to an @2x image.
-	bool Is2x(const string &path)
-	{
-		if(path.length() < 7)
-			return false;
-
-		size_t pos = path.length() - 7;
-		return (path[pos] == '@' && path[pos + 1] == '2' && path[pos + 2] == 'x');
-	}
-
-	// Determine whether the given path is to a swizzle mask.
-	bool IsSwizzleMask(const string &path, bool is2x)
-	{
-		if(path.length() < 7 || (is2x && path.length() < 10))
-			return false;
-
-		size_t pos = path.length() - (is2x ? 10 : 7);
-		return (path[pos] == '@' && path[pos + 1] == 's' && path[pos + 2] == 'w');
-	}
-
+	const set<string> SUPPORTED_EXTENSIONS{"png", "jpg", "jpeg", "jpe"};
 	// Check if the given character is a valid blending mode.
 	bool IsBlend(char c)
 	{
 		return (c == '-' || c == '~' || c == '+' || c == '=');
 	}
 
-	// Determine whether the given path or name is to a sprite for which a
-	// collision mask ought to be generated.
-	bool IsMasked(const string &path)
-	{
-		if(path.length() >= 5 && path.compare(0, 5, "ship/") == 0)
-			return true;
-		if(path.length() >= 9 && path.compare(0, 9, "asteroid/") == 0)
-			return true;
-
-		return false;
-	}
-
 	// Get the character index where the sprite name in the given path ends.
 	size_t NameEnd(const string &path)
 	{
-		// The path always ends in a three-letter extension, ".png" or ".jpg".
-		// In addition, 3 more characters may be taken up by an @2x label or a mask label.
-		bool is2x = Is2x(path);
-		size_t end = path.length() - (is2x ? 7 : 4) - (IsSwizzleMask(path, is2x) ? 3 : 0);
+		size_t lastDot = path.find_last_of('.');
+		if(lastDot == string::npos)
+			return 0;
+
+		// Get the name of the file, without the extension and the @2x or @sw label.
+		string base = path.substr(0, lastDot);
+		Format::ReplaceAll(base, "@2x", "");
+		Format::ReplaceAll(base, "@sw", "");
```

Comment:
I think we should a comment saying what this sw means (mb in the comment above?)

it used to be clear with the function names